Tempura/Navigation/NavigationActions.swift

```diff
@@ -191,52 +123,41 @@ public struct Hide: AnySideEffect {
   
   /// The side effect of the action, look into [Katana](https://github.com/BendingSpoons/katana-swift)
   /// to know what a `SideEffect` is.
-  public func sideEffect(_ context: AnySideEffectContext) throws {
+  public func anySideEffect(_ context: AnySideEffectContext) throws -> Any {
     guard let dependencies = context.anyDependencies as? NavigationProvider else { fatalError("DependenciesContainer must conform to `NavigationProvider`") }
     try await(dependencies.navigator.hide(self.identifierToHide, animated: self.animated, context: self.context, atomic: self.atomic))
+    return ()
   }
 }
 
-/// Old Hide action used to ask the `Navigator` to hide a specific screen
-/// identified by the `identifierToHide`.
-@available(*, deprecated, message: "With the new Katana 3.0 you should use the `Hide` SideEffect")
-struct HideLegacy: Action, ActionWithSideEffect {
-  var identifierToHide: RouteElementIdentifier
-  var animated: Bool
-  var context: Any?
-  var atomic: Bool
-  
-  /// Returns the new state after the action is dispatched.
-  public func updatedState(currentState: State) -> State {
-    return currentState
+// MARK: - Katana Helpers
+
+extension PartialStore {
+  public func dispatch<RSE: TempuraSideEffect>(_ dispatchable: RSE) -> Promise<Void> {
+    return self.anyDispatch(dispatchable).void
   }
-  
-  /// Initializes and return a Hide action.
-  public init(_ identifierToHide: RouteElementIdentifier, animated: Bool = false, context: Any? = nil, atomic: Bool = false) {
-    self.identifierToHide = identifierToHide
-    self.animated = animated
-    self.context = context
-    self.animated = animated
-    self.atomic = atomic
+
+  public func awaitDispatch<RSE: TempuraSideEffect>(_ dispatchable: RSE) throws {
+    return try await(self.dispatch(dispatchable))
   }
-  
-  /// Initializes and return a Hide action.
-  public init<K>(_ identifierToHide: K, animated: Bool = false, context: Any? = nil)
-    where K: RawRepresentable, K.RawValue == RouteElementIdentifier {
-      self.init(identifierToHide.rawValue, animated: animated, context: context)
+}
+
+extension SideEffectContext {
+  public func dispatch<RSE: TempuraSideEffect>(_ dispatchable: RSE) -> Promise<Void> {
+    return self.anyDispatch(dispatchable).void
   }
```

Comment:
Are any of these helpers used in this `NavigationActions.swift`? If not, I think we should move these in a separate file under `Utilities` as a file named `SideEffectContext+Utils.swift`.

Reply:
Do we really need these helpers? I feel like their place should be Katana, the only reason they are here I suppose is because of the `TempuraSideEffect` thing, my question is, do we need this `TempuraSideEffect` protocol?

Reply:
We can move them, I initially put them here because they are semantically related only to things inside this file

Reply:
Also yes, we need them to have a clean katana dispatch exposed to the app. Since tempura dispatchable inherit directly from `AnySideEffect` (and not `SideEffect`) the only dispatch that you can use to with them is `anyDispatch(_:) -> Promise<Any>` which is `poco elegante`.

Reply:
I got your point now! Then this is the right place for the extensions, but we need to rename `TempuraSideEffect` to `NavigationSideEffect` to express the fact that we expect a TempuraSideEffect to be used only in the context of navigation actions.